translator/util/sdkutil.go

```diff
@@ -90,17 +90,17 @@ func defaultECSRegion() string {
 func detectRegion(mode string, credsConfig map[string]string) (region string) {
 	region = SDKRegionWithCredsMap(mode, credsConfig)
 
-	// For ec2, fallback to metadata when no region info found in credential profile.
-	if region == "" && mode == config.ModeEC2 {
-		fmt.Println("I! Trying to detect region from ec2")
-		region = defaultEC2Region()
-	}
-
 	// try to get region from ecs metadata
 	if region == "" && mode == config.ModeEC2 {
 		fmt.Println("I! Trying to detect region from ecs")
 		region = defaultECSRegion()
 	}
+
+	// For ec2, fallback to metadata when no region info found in credential profile.
+	if region == "" && mode == config.ModeEC2 {
+		fmt.Println("I! Trying to detect region from ec2")
+		region = defaultEC2Region()
+	}
```

Comment:
For the concern, ECS Metadata only takes up to 4s based on the following log lines
```
2022/06/06 22:01:17 W! retry [0/3], unable to get http response from http://169.254.170.2/v2/metadata, error: unable to get response from http://169.254.170.2/v2/metadata, error: Get "http://169.254.170.2/v2/metadata": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2022/06/06 22:01:18 W! retry [1/3], unable to get http response from http://169.254.170.2/v2/metadata, error: unable to get response from http://169.254.170.2/v2/metadata, error: Get "http://169.254.170.2/v2/metadata": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2022/06/06 22:01:19 W! retry [2/3], unable to get http response from http://169.254.170.2/v2/metadata, error: unable to get response from http://169.254.170.2/v2/metadata, error: Get "http://169.254.170.2/v2/metadata": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2022/06/06 22:01:19 I! access ECS task metadata fail with response unable to get response from http://169.254.170.2/v2/metadata, error: Get "http://169.254.170.2/v2/metadata": context deadline exceeded (Client.Timeout exceeded while awaiting headers), assuming I'm not running in ECS.
```

I have three reasons why I prioritized ECS over EC2:
* CloudWatchAgent scraps only one ECS metadata endpoint which takes 4s while scraps [two IDMS endpoint which takes 8 second](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-dc70b68390b89e4d87ef2cec736f8587d0e736cecf1b032abf4891005ffa9432R102-R116)
* ECS metadata is more easier to scraped more than IDMS when consider hop limits (for both EC2 and Fargate)
* IMO, we would move away from EC2 soon and utilize more EKS and ECS consider scaling problems; therefore, EC2 is only a surface level problem when testing/developing/GA. Therefore, container is needed in a long-term run.

Let me know if that makes sense.

Reply:
> IMO, we would move away from EC2 soon and utilize more EKS and ECS consider scaling problems; therefore, EC2 is only a surface level problem when testing/developing/GA

I disagree. Do you have any data that would indicate that the percentage of customers using CloudWatch agent is trending more for EKS and ECS than regular EC2?

Reply:
> I disagree. Do you have any data that would indicate that the percentage of customers using CloudWatch agent is trending more for EKS and ECS than regular EC2?

No, that's why I only said its my personal opinion instead of reference any documents. However, I can get the data from CloudWatch Logs and Granite if it's good.

Reply:
This needs discussion internally. Even if it's your personal opinion, this PR would make it reality. I am not confident with approving this PR until we discuss design as a group.

Reply:
On top of the main issue, changing this behavior introduces unexpected warn log lines in the agent log. It introduces a false positive for customers who monitor the agent log with logs insights metrics. Although I don't think many people do that, I know some do, and this would impact their alarm thresholds. We should be able to resolve this issue without causing impact.

Reply:
Agree. If that's the case, revert back to original behavior which is outside of this PR scope. Thanks for reminding me of this.